### PR TITLE
Update all GHAs to their latest releases

### PR DESCRIPTION
**Pull request by dfarrell07**
See commit messages for details.

### CODE REVIEW

**Review by skitt — APPROVED** [review]

**Comment by dfarrell07** [comment]
Let's hold this until we merge https://github.com/submariner-io/submariner/pull/1751 so we can test if Dependabot can do SHA-based GHA updates.

**Comment by dfarrell07** [comment]
These updates were done by new automation.

**Pull request closed without merging**
